### PR TITLE
New if_else_match_braces_linter

### PR DESCRIPTION
**Pull request by MichaelChirico**
Part of #962

We might consider this one to be a default linter btw, it's basically part of the style guide.

### CODE REVIEW

**Comment by AshesITR** [comment]
In favor of making this a default linter, although I don't see explicit mention of this rule in https://style.tidyverse.org/syntax.html#control-flow
@jimhester WDYT? Should we lint

```r
if (test) a else {
  long_code
}

if (test) {
  long_code
} else a
```

by default?

**Comment by MichaelChirico** [comment]
FWIW styler agrees:

```
styler::style_text('if (test) a else {
  long_code
}

if (test) {
  long_code
} else a
')
# if (test) {
#   a
# } else {
#   long_code
# }

# if (test) {
#   long_code
# } else {
#   a
# }
```

**Comment by hadley** [comment]
Yeah this is definitely required by tidyverse style.

**Comment by MichaelChirico** [comment]
OK, done

File: tests/testthat/test-if_else_match_braces_linter.R [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  expect_lint("if (TRUE) 1", NULL, if_else_match_braces_linter())
+
+  lines_brace <- trim_some("
+    if (TRUE) {
+      1
+    } else {
+      2
+    }
+  ")
+  expect_lint(lines_brace, NULL, if_else_match_braces_linter())
```
Comment (AshesITR) [inline tests/testthat/test-if_else_match_braces_linter.R:12]: Just curious, why extract the `trim_some()` into a variable?

**Review by AshesITR — APPROVED** [review]
LGTM

File: tests/testthat/test-if_else_match_braces_linter.R (hunk 1 above)
Comment (MichaelChirico) [reply tests/testthat/test-if_else_match_braces_linter.R:12]: we do so internally with a preference erring on the side of assigning variables, especially in tests (where if the test fails, we can look at the intermediate output more easily).

i might not write it the same from scratch, but avoiding extra diffs from our internal source where possible

**Review by AshesITR — APPROVED** [review]

**Pull request merged**